doc/dev/testing.md

```diff
@@ -5,25 +5,33 @@ After [building Lean](../make/index.md) you can run all the tests using
 cd build/release
 make test ARGS=-j4
 ```
-
 Change the 4 to the maximum number of parallel tests you want to
 allow. The best choice is the number of CPU cores on your machine as
 the tests are mostly CPU bound.  You can find the number of processors
 on linux using `nproc` and on Windows it is the `NUMBER_OF_PROCESSORS`
 environment variable.
 
 You can run tests after [building a specific stage](bootstrap.md) by
-adding the `-C stageN` argument. The default when run as above is stage 1.  The
-Lean tests will automatically use that stage's corresponding Lean
+adding the `-C stageN` argument. The default when run as above is stage 1. The
+Lean tests will automatically use (but not build) that stage's corresponding Lean
```

Comment:
With `make test` they should, no? This caveat should be mentioned below for `ctest` instead

Reply:
> With make test they should, no? This caveat should be mentioned below for ctest instead

Not as far as I can tell:

```
~/build/lean/lean4 $ make -C build/release/stage1 test ARGS='-R wildcardAlt'
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird betreten
Running tests...
Test project /home/jojo/build/lean/lean4/build/release/stage1
    Start 571: leantest_wildcardAlt.lean
1/1 Test #571: leantest_wildcardAlt.lean ........   Passed    0.07 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.11 sec
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird verlassen

~/build/lean/lean4 $ sed -i -e 's,wildcard,wildcat,' src/Lean/Elab/Tactic/Induction.lean

~/build/lean/lean4 $ make -C build/release/stage1 test ARGS='-R wildcardAlt'
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird betreten
Running tests...
Test project /home/jojo/build/lean/lean4/build/release/stage1
    Start 571: leantest_wildcardAlt.lean
1/1 Test #571: leantest_wildcardAlt.lean ........   Passed    0.08 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.11 sec
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird verlassen

~/build/lean/lean4 $ make -C build/release/stage1 -j8
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird betreten
[  0%] Built target initialize
[  5%] Built target constructions
[ 27%] Built target kernel
[ 30%] Built target util
[ 44%] Built target library
[ 62%] Built target leanrt_initial-exec
[ 81%] Built target compiler
[ 82%] Built target shell
[100%] Built target leanrt
[100%] Built target leancpp
[    ] Building Lean/Elab/Tactic/Induction.lean
[    ] Building Lean/Elab/Tactic/Match.lean
[    ] Building ../build/release/stage1/lib/temp/Lean/Elab/Tactic/Induction.c
[    ] Building Lean/Elab/Tactic.lean
[    ] Building ../build/release/stage1/lib/temp/Lean/Elab/Tactic/Match.c
[    ] Building Lean/Elab.lean
[    ] Building ../build/release/stage1/lib/temp/Lean/Elab/Tactic.c
[    ] Building ../build/release/stage1/lib/temp/Lean/Elab.c
[    ] Building Lean.lean
[    ] Building ../build/release/stage1/lib/temp/Lean.c
[100%] Built target make_stdlib
[    ] Building /home/jojo/build/lean/lean4/build/release/stage1/lib/lean/libleanshared.so
[100%] Built target leanshared
[    ] Building /home/jojo/build/lean/lean4/build/release/stage1/bin/lean
[100%] Built target leanc
[100%] Built target lake
[100%] Built target lean
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird verlassen

~/build/lean/lean4 $ make -C build/release/stage1 test ARGS='-R wildcardAlt'
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird betreten
Running tests...
Test project /home/jojo/build/lean/lean4/build/release/stage1
    Start 571: leantest_wildcardAlt.lean
1/1 Test #571: leantest_wildcardAlt.lean ........***Failed    0.08 sec
--- wildcardAlt.lean.expected.out	2023-11-04 11:27:59.309845553 +0100
+++ wildcardAlt.lean.produced.out	2023-11-06 17:45:49.533216321 +0100
@@ -1,2 +1,2 @@
-wildcardAlt.lean:10:2-10:16: error: invalid occurrence of wildcard alternative, it must be the last alternative
-wildcardAlt.lean:15:2-15:16: error: invalid occurrence of wildcard alternative, it must be the last alternative
+wildcardAlt.lean:10:2-10:16: error: invalid occurrence of wildcat alternative, it must be the last alternative
+wildcardAlt.lean:15:2-15:16: error: invalid occurrence of wildcat alternative, it must be the last alternative
ERROR: file wildcardAlt.lean.produced.out does not match wildcardAlt.lean.expected.out


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.11 sec

The following tests FAILED:
	571 - leantest_wildcardAlt.lean (Failed)
Errors while running CTest
make: *** [Makefile:71: test] Fehler 8
make: Verzeichnis „/home/jojo/build/lean/lean4/build/release/stage1“ wird verlassen
```

Reply:
@Kha does this need more investigation, or should we just merge the doc update as is?

Reply:
I’ll merge this in a few days if nobody complains, we can still refine on `master`.

Reply:
No idea, I never have :-)

My point is simply that `make test` in `build/release/stage1` doesn’t  build stage1 itself, which may or may not be surprising to some, so worth documenting? A simple

> Note that running the tests with `make test` does not build the `stage1` compiler.

would also work.

Reply:
Ok, I'm not opposed to either wording. I just think if there's no interesting use case to ever `make` in stage1, I'd rather not mention it at all.